### DISCUSSION TITLE
Conversion older ggml code to latest version

### DISCUSSION THREAD

**Discussion by HoiV** [ggml-org/ggml#738] [Q&A] [answered]
Hi, to learn more from GGML, I am trying to convert some older apps to run with the latest GGML lib. 

In doing so, I ran across this type of construct: 
        inpL = ggml_mul_mat(ctx0, inpL, sum);

        // normalizer
        struct ggml_tensor *sqr1 = ggml_sqr(ctx0, inpL);
        struct ggml_tensor *sum1 = ggml_sum(ctx0, sqr1);
        struct ggml_tensor *sqrt1 = ggml_sqrt(ctx0, sum1);
        inpL = ggml_scale(ctx0, inpL, ggml_div(ctx0, ggml_new_f32(ctx0, 1.0f), sqrt1));

Since ggml_scale() only accepts a float for the 3rd parameter, what is the best way to get the right value from:

"ggml_div(ctx0, ggml_new_f32(ctx0, 1.0f), sqrt1)"?

Thank you in advance.

**Comment by slaren** [comment, marked as answer]
`ggml_mul` now supports broadcasting in all the dimensions, so you can use it in place of `ggml_scale`.
